core/trie/proof_test.go

```diff
@@ -869,3 +918,244 @@ func TestVerifyRangeProof(t *testing.T) {
 		require.True(t, verif)
 	})
 }
+
+func TestMergeProofPaths(t *testing.T) {
+	t.Run("3Key Trie no duplicates and all values exist in merged path", func(t *testing.T) {
+		tri := build3KeyTrie(t)
+
+		zeroFeltBytes := new(felt.Felt).SetUint64(0).Bytes()
+		zeroLeafkey := trie.NewKey(251, zeroFeltBytes[:])
+
+		twoFeltBytes := new(felt.Felt).SetUint64(2).Bytes()
+		twoLeafkey := trie.NewKey(251, twoFeltBytes[:])
+
+		proofKeys := [2]*trie.Key{&zeroLeafkey, &twoLeafkey}
+
+		proofs, err := trie.GetBoundaryProofs(proofKeys[0], proofKeys[1], tri)
+		require.NoError(t, err)
+
+		mergedProofs, _, err := trie.MergeProofPaths(proofs[0], proofs[1], crypto.Pedersen)
+		require.NoError(t, err)
+
+		containsLeft := containsAll(mergedProofs, proofs[0])
+		require.True(t, containsLeft)
+		containsRight := containsAll(mergedProofs, proofs[1])
+		require.True(t, containsRight)
+		checkDuplicates := noDuplicates(mergedProofs)
+		require.True(t, checkDuplicates)
```

Comment:
nit:
require.True(t, containsAll(mergedProofs, proofs[0]))
		require.True(t, containsAll(mergedProofs, proofs[1]))
		require.True(t, noDuplicates(mergedProofs))

Reply:
same for all tests below